src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1291,6 +1313,51 @@ public void TestConflictResolverExceptionsNoneMatchDBThrown()
             }
         }
 
+        [Fact]
+        public void TestConflictResolverReturningBlob()
+        {
+            var returnRemoteDoc = true;
+            TestConflictResolverWins(returnRemoteDoc);
+            TestConflictResolverWins(!returnRemoteDoc);
+
+            //return new doc with a blob object
+            CreateReplicationConflict();
+
+            var config = CreateConfig(false, true, false);
+
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                var evilByteArray = new byte[] { 6, 6, 6 };
+
+                var doc = new MutableDocument();
+                doc.SetBlob("blob", new Blob("text/plaintext", evilByteArray));
+                return doc;
+            });
+
+            RunReplication(config, 0, 0);
+
+            using (var doc = Db.GetDocument("doc1")) {
+                doc.GetBlob("blob")?.Content.Should().ContainInOrder(new byte[] { 6, 6, 6 });
+            }
+        }
+
+        [Fact]
+        public void TestConflictResolverReturningBlobFromDifferentDB()
+        {
+            var evilByteArray = new byte[] { 6, 6, 6 };
+            using (var thirdDb = new Database("different_db"))
+            using (var doc1 = new MutableDocument("doc1")) {
+                doc1.SetBlob("blob", new Blob("text/plaintext", evilByteArray));
+                thirdDb.Save(doc1);
+
+                var differentDbResolver = new TestConflictResolver((conflict) => doc1);
```

Comment:
Still this is returning a document from another database.  You need to create a new document *inside the conflict resolver* and set a blob on it that comes from a document from another database.